### ISSUE TITLE
3D Particles, Animated Directional Velocity: Local Direction Option and Min<>Max Per Axis

### ISSUE THREAD

**Issue by LordMcMutton** [godotengine/godot-proposals#11329]
### Describe the project you are working on

Specifically, I had a lovely little particle system set up for a dust trail that kicks up when the character slides.

It generated a cloud in global space that would leave a trail behind when the character was moving, and I had the Animated Directional Velocity set up such that the cloud widened over time before dissipating 

### Describe the problem or limitation you are having in your project

Issue 1, Directional Velocity and Local Coords

Once I applied this to the character to see it in action, to my dismay, the cloud wasn't heeding the particle emitter's rotation. Turns out that without Local Coords checked, the Animated Velocity options only work in global space.

Unfortunately, if I check Local Coords, then it stops working as a trail, as the particles simply stick to the emitter

Issue 2, Min<>Max applied to every axis

When making the cloud, I wanted to have particles kick off to both sides, so I set up Directional Velocity's X Curve and then the Min<>Max to -1<>1. This had the desired effect.

Then I wanted to have the particles float upwards as they dissipated, so I set up Directional Velocity's Y Curve. To my dismay, the particles were floating upward AND falling downward. Min<>Max affects every axis equally, so what I wanted to do wasn't possible.

I was able to use Gravity to get something like what I wanted, but I think being able to execute such things with far more intent would make for a more robust system

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Issue 1, Directional Velocity and Local Coords

In terms of UI, this is as easy as setting a checkbox within Directional Velocity that would allow you to make the Curves apply in local space while simulating the particles in global space

An alternative is one I've seen proposed before: Emitting X particles per unit of distance traveled. That would also work for my purposes, but I feel like it might be a more limited option than the other one.

Issue 2, Min<>Max applied to every axis

The Min<>Max UI element is nifty, but I'd suggest getting rid of it in favor of having two Curve Textures per axis: Min Curve and Max Curve. After setting the two up, the particles would take the path at any point between the two sets of values, much like it does already, but with more fine control

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Basically covered in the previous section: Check the Local Space option under Directional Velocity to have the Curves apply according to the Emitter's facing direction, and set up each Axes' Min and Max Curves to dictate exactly how the particles will move

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It would be used often enough

### Is there a reason why this should be core and not an add-on in the asset library?

This addresses limitations with the current setup- there's no reason why it shouldn't be more robust

**Comment by Calinou** [comment]
> In terms of UI, this is as easy as setting a checkbox within Directional Velocity that would allow you to make the Curves apply in local space while simulating the particles in global space

This may be better covered by a universal toggle that affects all kinds of rotation: https://github.com/godotengine/godot-proposals/issues/4391

**Comment by LordMcMutton** [comment]
Since that proposal is essentially the same thing, should I just remove the Local Directional Velocity stuff entirely and thumbs up that one?

**Comment by Calinou** [comment]
> Since that proposal is essentially the same thing, should I just remove the Local Directional Velocity stuff entirely and thumbs up that one?

Yes 🙂

**Comment by QbieShay** [comment]
Chiming in on this: I think in general it's a good idea, but I'm a little concerned of the exploding amount of shader variants if every single minmax property can be 1d or 3d. Need input from lower level rendering people ( @Calinou , @clayjohn , @DarioSamo )
